ivy/array/losses.py

```diff
@@ -1,8 +1,37 @@
 # global
 import abc
+from typing import Optional, Union
 
-# ToDo: implement all methods here as public instance methods
+# local
+import ivy
 
 
 class ArrayWithLosses(abc.ABC):
+    def cross_entropy(
+        self: ivy.Array,
+        pred: Union[ivy.Array, ivy.NativeArray],
+        axis: Optional[int] = -1,
+        epsilon: Optional[float] = 1e-7,
+        *,
+        out: Optional[ivy.Array] = None
+    ) -> ivy.Array:
+        return ivy.cross_entropy(self, pred, axis=axis, epsilon=epsilon, out=out)
```

Comment:
`self._data` should be passed in rather than `self` for all instance methods. This is a small detail, but is better for consistency with the other functions. This point is explained [here](https://github.com/unifyai/ivy/blob/e7712dffabe126fc795d52c2fd0d5b05a91b064f/docs/partial_source/deep_dive/4_arrays.rst#ivy-array).